_posts/2022-01-10-reassign-partitions.md

```diff
@@ -0,0 +1,363 @@
+```
+Now you can create a `reassignment.json` file and copy this proposed `reassignment.json` data to use in the `reassignment.json` file. You can also create/alter the `reassignment.json` data in the file.
+
+### Scaling down the cluster
+
+Coming to the last step of this example, Here we will see how we use the generated `reassignment.json` to get the reassignment done by the reassignment partition tool.
+
+Since we have copied the `reassignment.json` data into the `reassignment.json` file in the previous step, now we can use it for our next step.
+
+Let's copy the `reassignment.json` file to the interactive pod container.
+
+```sh
+kubectl cp reassignment.json <INTERACTIVE-POD-NAME>:/tmp/reassignment.json
+```
+
+Now we'll start a shell process inside the interactive pod container to run our Kafka bin script.
+```sh
+kubectl exec -ti <INTERACTIVE-POD-NAME> /bin/bash
+```
+
+So let's run the `kafka-reassign-partitions.sh` script now to start the partition reassignment 
+
+```sh
+bin/kafka-reassign-partitions.sh --bootstrap-server <CLUSTER-NAME>-kafka-bootstrap:9092 \
+ --reassignment-json-file /tmp/reassignment.json \ 
+ --execute
+```
+
+You can use the `--verify` mode to check if the partition reassignment is done or if it is still running.
+
+```sh
+bin/kafka-reassign-partitions.sh --bootstrap-server <CLUSTER-NAME>-kafka-bootstrap:9092 \
+  --reassignment-json-file /tmp/reassignment.json \
+  --verify
+```
+The `--verify` mode reports if the assignment is done or not.
+
+```shell
+Status of partition reassignment:
+Reassignment of partition my-topic-1-0 is complete.
+Reassignment of partition my-topic-1-1 is complete.
+Reassignment of partition my-topic-1-2 is complete.
+Reassignment of partition my-topic-1-3 is complete.
+Reassignment of partition my-topic-1-4 is complete.
+Reassignment of partition my-topic-1-5 is complete.
+Reassignment of partition my-topic-1-6 is complete.
+Reassignment of partition my-topic-1-7 is complete.
+Reassignment of partition my-topic-1-8 is complete.
+Reassignment of partition my-topic-1-9 is complete.
+
+Clearing broker-level throttles on brokers 0,1,2,3,4
+Clearing topic-level throttles on topic my-topic-1  
+```
+
+When the partition reassignment is complete, the scaled down brokers will have no assigned partitions and can be removed safely. You can check it by looking at the partition details of the reassigned topic.
```

Comment:
`the scaled down brokers` -> `the brokers we want to scale down`. At this point, these brokers are still there.